src/Features/LanguageServer/Protocol/DefaultCapabilitiesProvider.cs

```diff
@@ -106,6 +106,12 @@ public ServerCapabilities GetCapabilities(ClientCapabilities clientCapabilities)
                 WorkDoneProgress = false,
             };
 
+            capabilities.InlayHintOptions = new InlayHintOptions
+            {
+                ResolveProvider = true,
```

Comment:
Eventually (and fine not to do yet, maybe wait until Shen's work is fully in) - we should set the server capabilities based on whether or not inlay hints are enabled.  We'd have to do it dynamically though and react to option changes to send the client/registercapability request if they change.